homeassistant/components/recorder/__init__.py

```diff
@@ -436,14 +488,43 @@ def async_connection_success(self):
 
     @callback
     def _async_recorder_ready(self):
-        """Mark recorder ready."""
+        """Finish start and mark recorder ready."""
+        self._async_setup_periodic_tasks()
         self.async_recorder_ready.set()
 
     @callback
     def async_purge(self, now):
         """Trigger the purge."""
         self.queue.put(PurgeTask(self.keep_days, repack=False, apply_filter=False))
 
+    @callback
+    def async_daily_statistics(self, now):
+        """Trigger the daily statistics run."""
+        start = statistics.get_start_time("daily")
+        self.queue.put(StatisticsTask("daily", start))
+
+    @callback
+    def async_hourly_statistics(self, now):
+        """Trigger the hourly statistics run."""
+        start = statistics.get_start_time("hourly")
+        self.queue.put(StatisticsTask("hourly", start))
+
+    def _async_setup_periodic_tasks(self):
+        """Prepare periodic tasks."""
+        if self.auto_purge:
+            # Purge every night at 4:12am
+            async_track_time_change(
+                self.hass, self.async_purge, hour=4, minute=12, second=0
+            )
+        # Compile daily statistics every night at 12:12am
+        async_track_time_change(
+            self.hass, self.async_daily_statistics, hour=12, minute=12, second=0
+        )
+        # Compile hourly statistics every hour at *:12
```

Comment:
This will overlap with the daily stats + the hourly stats this might be based on could be not done yet ?

Reply:
Right. I'm not sure if we should have any daily stats at all, hourly might be enough?

Reply:
Yeah works for me.

Reply:
We can always expand later.